test/cppsuite/test_harness/workload/workload_validation.h

```diff
@@ -179,89 +199,72 @@ class workload_validation {
 
     /* Update the data model. */
     void
-    update_data_model(const tracking_operation &operation, const uint64_t collection_id,
-      const char *key, const char *value, database &database)
+    update_data_model(const tracking_operation &operation, validation_collection &collection,
+      const uint64_t collection_id, const char *key, const char *value)
     {
-        collection &collection = database.get_collection(collection_id);
-        switch (operation) {
-        case tracking_operation::DELETE_KEY:
-            /*
-             * Operations are parsed from the oldest to the most recent one. It is safe to assume
-             * the key has been inserted previously in an existing collection and can be safely
-             * deleted.
-             */
-            collection.delete_record(key);
-            break;
-        case tracking_operation::INSERT: {
-            /*
-             * Keys are unique, it is safe to assume the key has not been encountered before.
-             */
-            collection.insert_record(key, value);
-            break;
-        }
-        case tracking_operation::UPDATE:
-            collection.update_record(key, value);
-            break;
-        default:
-            testutil_die(DEBUG_ERROR, "Unexpected operation in the tracking table: %d",
+        if (operation == tracking_operation::DELETE_KEY) {
```

Comment:
Converted from switch case to ifs here as we needed brackets on a few of them so I think its cleaner this way.